src/NodeApi/JSDispatcherQueue.cs

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Hermes.Example;
+
+public delegate void JSTypedEventHandler<TSender, TResult>(TSender sender, TResult args);
+
+public sealed class DispatcherQueueShutdownStartingEventArgs : EventArgs
+{
+    private Func<JSDispatcherQueueDeferral> _getDeferral;
+
+    internal DispatcherQueueShutdownStartingEventArgs(Func<JSDispatcherQueueDeferral> getDeferral)
+        => _getDeferral = getDeferral;
+
+    public JSDispatcherQueueDeferral GetDeferral() => _getDeferral();
```

Comment:
Can it just return `IDisposable` while keeping the deferral type internal?

Reply:
Agree. It makes the public surface smaller. Changed.